sdcm/nemesis.py

```diff
@@ -466,6 +467,29 @@ def disrupt_rolling_restart_cluster(self):
             raise UnsupportedNemesis('RollingRestartCluster is supported only for kubernetes backends')
         self.cluster.rollout_restart()
 
+    def disrupt_toggle_authenticator_type(self):
+        """
+        If prepare_saslauthd is enabled, saslauthd and ldap environment will be prepared for
+        using SaslauthdAuthenticator. We have same account (cassandra) for SaslauthdAuthenticator
+        and PasswordAuthenticator, so it can be toggled smoothly without effect to c-s workloads.
+
+        It's only support to switch from PasswordAuthenticator, the authenticator will be reset
+        back in the end of nemesis.
+        """
+        self._set_current_disruption('ToggleAuthenticatorType %s' % self.target_node)
+        # Switch to com.scylladb.auth.SaslauthdAuthenticator
+        if self.cluster.params.get('prepare_saslauthd') and 'PasswordAuthenticator' in self.cluster.params.get('authenticator'):
```

Comment:
As we discuss let's change this to check the current configuration (not the params) and if it's sasl change it to Password and vise versa.

Reply:
Fixed.